pkg/azurefile/nodeserver.go

```diff
@@ -95,6 +96,35 @@ func (d *Driver) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 				return nil, status.Errorf(codes.InvalidArgument, fmt.Sprintf("invalid mountPermissions %s", perm))
 			}
 		}
+
+		// get runtime class using pod info from volume context
+		runtimeClass, err := getRuntimeClassForPod(ctx, d.kubeconfig, context[podNameField], context[podNamespaceField])
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get runtime class for pod %s/%s: %v", context[podNamespaceField], context[podNameField], err)
+		}
+		klog.V(2).Infof("NodePublishVolume: volume(%s) mount on %s with runtimeClass %s", volumeID, target, runtimeClass)
+		if isConfidentialRuntimeClass(runtimeClass) {
```

Comment:
@andyzhangx So here we're enabling the conf pods path based on fixed runtime class names. However customers could create different runtime classes that refer to the `kata-cc` handler. Should we resolve the final handler name then or are you happy with this?